pkg/cmd/cli/cmd/login/helpers.go

```diff
@@ -0,0 +1,55 @@
+package login
+
+import (
+	"bytes"
+
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/user/api"
+
+	"k8s.io/kubernetes/pkg/client/restclient"
+	clientcmdapi "k8s.io/kubernetes/pkg/client/unversioned/clientcmd/api"
+	"k8s.io/kubernetes/pkg/util/sets"
+)
+
+// getMatchingClusters examines the kubeconfig for all clusters that point to the same server
+func getMatchingClusters(clientConfig restclient.Config, kubeconfig clientcmdapi.Config) sets.String {
+	ret := sets.String{}
+
+	for key, cluster := range kubeconfig.Clusters {
+		if (cluster.Server == clientConfig.Host) && (cluster.InsecureSkipTLSVerify == clientConfig.Insecure) && (cluster.CertificateAuthority == clientConfig.CAFile) && (bytes.Compare(cluster.CertificateAuthorityData, clientConfig.CAData) == 0) {
+			ret.Insert(key)
+		}
+	}
+
+	return ret
+}
+
+type existingClientCertificate struct {
+	cAFile string
+	cAData []byte
+}
+
+// findExistingClientCA returns *either* the existing client CA file *or* data for
+// a given host, and true if it exists in the given config
+func findExistingClientCA(host string, kubeconfig clientcmdapi.Config) (*existingClientCertificate, bool) {
+	for _, cluster := range kubeconfig.Clusters {
+		if cluster.Server == host {
+			if len(cluster.CertificateAuthority) > 0 {
```

Comment:
I'm ok with removing the struct but I wanted `findExistingClientCA` to only return *one* of `CertificateAuthority` or `CertificateAuthorityData`. Having both in the client config is an error and if somehow both are present in a cluster entry we read here, your suggestion doesn't protect us from that problem. Ok with me keeping these `if`s?